lbr_demos/lbr_demos_fri_ros2_cpp/src/cartesian_pose_node.cpp

```diff
@@ -0,0 +1,246 @@
+#include "lbr_fri_msgs/msg/lbr_state.hpp"
+#include "friClientIf.h"
+
+using std::placeholders::_1;
+
+class CartesianPoseNode:public rclcpp::Node
+{
+  private:
+    rclcpp::Publisher<lbr_fri_msgs::msg::LBRPositionCommand>::SharedPtr joint_position_publisher_;
+    rclcpp::Subscription<lbr_fri_msgs::msg::LBRState>::SharedPtr joint_position_subscriber_;
+    rclcpp::Publisher<geometry_msgs::msg::Pose>::SharedPtr cartesian_pose_publisher_;
+    rclcpp::Subscription<geometry_msgs::msg::Pose>::SharedPtr cartesian_pose_subscriber_;
+
+    lbr_fri_msgs::msg::LBRState current_robot_state_; // robot state, including joint positions
+    geometry_msgs::msg::Pose current_cartesian_position_; // current cartesian pose of robot
+
+    KDL::Chain chain_; // robot kinematics chain exetracted from robot URDF file
+
+  private:
+    /**
+     * @function: callback function for Joint Position Subscriber
+     * @param msg joint position of the robot
+    */
+    void joint_position_sub_callback(const lbr_fri_msgs::msg::LBRState& msg)
+    {
+      current_robot_state_ = msg;
+            
+      double joint_position[7];
+      for(int i = 0; i < 7; i++)
+      {
+        joint_position[i] = current_robot_state_.measured_joint_position[i]; 
+      }
+      current_cartesian_position_ = computeForwardKinematics(joint_position);
+      cartesian_pose_publisher_->publish(current_cartesian_position_);
+
+      return;
+    }
+
+    /**
+     * @function: callback function for Cartesian Pose Subscriber
+     * @param msg cartesian pose command
+    */
+    void cartesian_pose_sub_callback(const geometry_msgs::msg::Pose& msg)
+    {
+      if(current_robot_state_.session_state == KUKA::FRI::COMMANDING_ACTIVE) 
+      {
+        unsigned int joint_number = chain_.getNrOfJoints(); // for kuka robot, 7 joints 
+        KDL::JntArray current_joint_positions(joint_number);
+        lbr_fri_msgs::msg::LBRPositionCommand joint_position_command;
+
+        for(unsigned int i = 0; i < joint_number; i++)
+        {
+          current_joint_positions(i) = current_robot_state_.measured_joint_position[i];
+        }
+
+        joint_position_command = computeInverseKinematics(msg, current_joint_positions);
+        joint_position_publisher_->publish(joint_position_command);
+      }
+
+      return;
+    }
+
+  public:
+    CartesianPoseNode():Node("cartesian_pose_node")
+    {
+      this->declare_parameter<std::string>("robot_name", "iiwa7"); // default name "iiwa7"
+
+      std::string robot_name;
+      this->get_parameter("robot_name", robot_name);
+
+      if(robot_name != "iiwa7" && robot_name != "iiwa14" && 
+         robot_name != "med7" && robot_name != "med14")
+      {
+        std::cerr << "Wrong robot name. " << std::endl;
+      }
+
+      // get the path of urdf file 
+      std::filesystem::path current_working_directory = std::filesystem::current_path();
+      std::string urdf_file_path = current_working_directory.string() 
```

Comment:
getting the current path might not work in general

Reply:
Hello, @mhubii,  thank you very much for your suggestion and QoS documentation, I learn a lot from it. I modify my code following your advice today and then upload it again after I test it.

Reply:
awesome, happy to merge once the `/robot_description` is loaded from topic / parameter and `.urdf` files are removed